jira/resilientsession.py

```diff
@@ -113,11 +113,14 @@ def __recoverable(
                 f"Got ConnectionError [{response}] errno:{response.errno} on {request} {url}\n{vars(response)}\n{response.__dict__}"
             )
         if isinstance(response, Response):
-            if response.status_code in [502, 503, 504, 401]:
-                # 401 UNAUTHORIZED still randomly returned by Atlassian Cloud as of 2017-01-16
+            if response.status_code in [429]:
+                rate_limit_remaining = response.headers["X-RateLimit-Remaining"]
                 msg = f"{response.status_code} {response.reason}"
-                # 2019-07-25: Disabled recovery for codes above^
-                return False
+                logging.warning(
+                    f"""Request rate limited by Jira: number of tokens remaining {rate_limit_remaining}. Consider adding exemption for the user as explained in
+                    https://confluence.atlassian.com/adminjiraserver/improving-instance-stability-with-rate-limiting-983794911.html
```

Comment:
Actually was just reading the page, and rate_limit_remaining might not be useful here
if we are rate limit that means we have hit the limit? So we might just be logging 0...

Maybe instead to we should log the `X-RateLimit-Limit` and `retry-after` values which represent how many tokens we have and how long we need to wait before the next retry

Reply:
I think the most interesting headers in this scenario would be `X-RateLimit-FillRate` and `X-RateLimit-Interval-Seconds`, so that a user would get the sense of how often tokens are issued and how many tokens are issued each time, to better understand how to perform and possibly configure retries accordingly. `X-RateLimit-Limit` is the maximum value of tokens that you can accumulate over time, it's still good info but I believe the former ones are more actionable. Anyhow I'll add all of the above.

And yes, is my understanding as well that in a throttled request scenario the value of `X-RateLimit-Remaining` would be 0 so I'll remove it.